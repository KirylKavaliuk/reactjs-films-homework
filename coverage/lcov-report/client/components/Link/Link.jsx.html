<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Link/Link.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Link</a> Link.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">29x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">29x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">71x</span>
<span class="cline-any cline-yes">71x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">71x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
&nbsp;
import classNames from 'classnames';
&nbsp;
import { Params, getSection } from 'utils/url';
&nbsp;
import styles from './Link.scss';
&nbsp;
export default class _Link extends Component {
  state = {
    to: '',
  }
&nbsp;
  componentDidMount() {
    const to = this.props.to || getSection();
&nbsp;
    this.setState({ to });
  }
&nbsp;
  onHoverHandler = () =&gt; {
    const { to, params, clearParams } = this.props;
&nbsp;
    const section = to || getSection();
    const search = new Params()
      .add(params)
      .remove(clearParams)
      .toString();
&nbsp;
    this.setState({ to: `${section}${search}` });
  }
&nbsp;
  render() {
    const {
      onClick,
      className,
      children,
    } = this.props;
    const { to } = this.state;
&nbsp;
    return (
      &lt;Link
        onMouseEnter={ this.onHoverHandler }
        onClick={ onClick }
        to={ to }
        className={ classNames(styles.link, className) }
      &gt;
        { children }
      &lt;/Link&gt;
    );
  }
}
&nbsp;
_Link.defaultProps = {
  params: {},
  clearParams: [],
  to: '',
  className: '',
};
&nbsp;
_Link.propTypes = {
  params: PropTypes.object,
  clearParams: PropTypes.array,
  to: PropTypes.string,
  className: PropTypes.string,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
