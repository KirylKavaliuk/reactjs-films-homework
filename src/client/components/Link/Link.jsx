import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import classNames from 'classnames';

import { Params, getSection } from 'utils/url';

import styles from './Link.scss';

export default class _Link extends Component {
  state = {
    to: '',
  }

  componentDidMount() {
    const to = this.props.to || getSection();

    this.setState({ to });
  }

  onHoverHandler = () => {
    const {
      to, params, clearParams,
    } = this.props;

    const section = to || getSection();
    const search = new Params()
      .add(params)
      .remove(clearParams)
      .toString();

    this.setState({ to: `${section}${search}` });
  }

  render() {
    const {
      onClick, className, children,
    } = this.props;
    const { to } = this.state;
    const classes = classNames(styles.link, className);

    return (
      <Link
        onMouseEnter={ this.onHoverHandler }
        onClick={ onClick }
        to={ to }
        className={ classes }
      >
        { children }
      </Link>
    );
  }
}

_Link.defaultProps = {
  params: {},
  clearParams: [],
  to: '',
  className: '',
};

_Link.propTypes = {
  params: PropTypes.object,
  clearParams: PropTypes.array,
  to: PropTypes.string,
  className: PropTypes.string,
};
