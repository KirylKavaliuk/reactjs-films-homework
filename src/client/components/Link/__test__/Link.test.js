import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Link from '../Link';

it('<Link/>', () => {
  const treeWithPath = renderer
    .create(
      <MemoryRouter>
        <Link to='/section'/>
      </MemoryRouter>,
    ).toJSON();

  treeWithPath.props.onMouseEnter();
  expect(treeWithPath).not.toBeNull();
  expect(treeWithPath).toMatchSnapshot();

  const treeWithoutPath = renderer
    .create(
      <MemoryRouter>
        <Link/>
      </MemoryRouter>,
    ).toJSON();

  treeWithoutPath.props.onMouseEnter();
  expect(treeWithoutPath).not.toBeNull();
  expect(treeWithoutPath).toMatchSnapshot();
});
