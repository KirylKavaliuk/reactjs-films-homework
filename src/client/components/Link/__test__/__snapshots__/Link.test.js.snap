// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link/> 1`] = `
<a
  className="Link__link___-IU0S"
  href="/section"
  onClick={[Function]}
  onMouseEnter={[Function]}
/>
`;

exports[`<Link/> 2`] = `
<a
  className="Link__link___-IU0S"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
/>
`;
